pennylane/optimize/gradient_descent.py

```diff
@@ -129,7 +129,12 @@ def compute_grad(objective_fn, args, kwargs, grad_fn=None):
         grad = g(*args, **kwargs)
         forward = getattr(g, "forward", None)
 
-        if len(args) == 1:
+        num_trainable_args = 0
+        for arg in args:
+            if getattr(arg, "requires_grad", True):
+                num_trainable_args += 1
```

Comment:
@antalszava you might be able to use the new
```python
num_trainable_args = len(qml.math.get_trainable_indices(args))
```

functionality I just merged into the math module!

Reply:
(optional, though)

Reply:
This didn't seem to have worked for some tests :thinking:

Reply:
No worries! We'll need to remember to update this in the future, if `requires_grad=False` by default

Reply:
The issue seems to be that if the argument is not a PennyLane NumPy tensor, then it'll be considered to belong to the "numpy" interface (instead of the "autgrad" interface), and all "numpy" arguments automatically evaluate as `False` when calling `utils.requires_grad(arg)`. So, whenever an `arg` is simply a float `get_trainable_indices` would consider it non-trainiable, while that seems to not be what's wanted here.